doc/design/model_zoo.md

```diff
@@ -60,97 +60,99 @@ In SQLFlow SQL grammar, the identifiers after `TRAIN`, `USING` and `INTO` have d
 
-It is necessary to have the model ID so users can refer to the trained model when they want to use it.  Suppose that the user typed the prediction SQL statement at the beginning of this document. SQLFlow server will convert it into a submitter program and run it with the Docker image used to train the model. Therefore, the Docker image ID is also required. The model parameter path allows the prediction submitter program to locate and load the trained models.  The data converter helps the prediction submitter to use the conversion rules consistent with the ones used when training.
+### Use the Model Definition in the Docker Image
 
-It is necessary to record the content or the MD5 hash of the training submitter program in the model zoo table for experiment management. Please be aware that the training submitter encodes all three categories of hyperparameters, as listed in the above sections.  Suppose that the analyst re-trains the model with different hyperparameter settings, the training submitter changes accordingly, and SQLFlow should be able to remind the analyst to either uses a new model ID or overwrites the existing row in the model zoo table.
+For a SELECT program using the SQLFlow syntax extension, the SQLFlow server converts it into a [workflow](workflow.md) and submit the workflow job to a workflow engine like Argo/Tekton on Kubernetes. Each step in the workflow is one SQL statement.
 
-We recommend reusing the DBMS configured as the data source of SQLFlow for storing model zoo.  Following this recommendation, users can query the trained models using SQL.  For example, the following SQL statement lists all models trained by `an_analyst`:
+By default, we use a default Docker image to run the training, predicting or explaining job. The default Docker image contains pre-made Tensorflow estimator models, Keras models defined in [sqlflow_models repo](https://github.com/sql-machine-learning/models) and XGBoost. To use a custom model definition Docker image, write SQL statements mentioned above:
 
 ```sql
-SELECT * FROM sqlflow.trained_models WHERE creator="an_analyst"
+SELECT * FROM employee WHERE onboard_year < 2019
+TO TRAIN a_data_scientist/regressors:v0.2/MyDNNRegressor
+WITH model.hidden_units=[1024,256],
+LABEL salary
+INTO my_first_model;
 ```
 
-The following statement queries all model zoos used by `an_analyst` to train his/her models.
+Then the generated workflow will use the Docker image `a_data_scientist/regressors:v0.2` to run the statement. In this step, SQLFlow will generate the training Python program inside the Docker container and execute it. Once the job completes, **the trained model together with hyperparameters, Docker image, evaluation result and the SQL statement used will be saved.** So in one trained model, we can have:
 
-```sql
-SELECT DISTINCT model_zoo FROM sqlflow.trained_models WHERE creator="an_analyst"
-```
+1. Trained model weights.
```

Comment:
I think we need also store the generated transform code built using `feature column` or `keras preprocessing layers` in the trained model.
The generated code not only contains the data transform logic but also the data analysis result such as `bucketize boundaries`, `hash_bucket_size`, `min/max` and so on.